autopush/router/apnsrouter.py

```diff
@@ -147,29 +147,32 @@ def _route(self, notification, router_data):
         apns_id = str(uuid.uuid4()).lower()
         # APNs may force close a connection on us without warning.
         # if that happens, retry the message.
-        success = False
         try:
             apns_client.send(router_token=router_token, payload=payload,
                              apns_id=apns_id)
-            success = True
-        except ConnectionError:
-            self.metrics.increment("notification.bridge.connection.error",
-                                   tags=make_tags(
-                                       self._base_tags,
-                                       application=rel_channel,
-                                       reason="connection_error"))
-        except (HTTP20Error, socket.error):
+        except Exception as e:
+            # We sometimes see strange errors around sending push notifications
+            # to APNS. We get reports that after a new deployment things work,
+            # but then after a week or so, messages across the APNS bridge
+            # start to fail. The connections appear to be working correctly,
+            # so we don't think that this is a problem related to how we're
+            # connecting.
+            if isinstance(e, ConnectionError):
+                reason = "connection_error"
+            elif isinstance(e, (HTTP20Error, socket.error)):
+                reason = "http2_error"
+            else:
+                reason = "unknown"
             self.metrics.increment("notification.bridge.connection.error",
                                    tags=make_tags(self._base_tags,
                                                   application=rel_channel,
-                                                  reason="http2_error"))
-        if not success:
+                                                  reason=reason))
             raise RouterException(
-                "Server error",
+                str(e),
```

Comment:
I know you said you wanted to add a prefix here

Reply:
Yeah, dropped that. There's a prefix that's already appended so it became redundant. I noticed it when I was fixing the tests.